clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.cpp

```diff
@@ -519,7 +846,39 @@ Value LowerFunction::rewriteCallOp(const LowerFunctionInfo &CallInfo,
         }
       }
 
-      llvm_unreachable("NYI");
+      // If coercing a fixed vector from a scalable vector for ABI
+      // compatibility, and the types match, use the llvm.vector.extract
+      // intrinsic to perform the conversion.
+      if (::cir::MissingFeatures::vectorType()) {
+        llvm_unreachable("NYI");
+      }
+
+      // FIXME(cir): Use return value slot here.
+      Value RetVal = callOp.getResult();
+      // TODO(cir): Check for volatile return values.
+
+      // NOTE(cir): If the function returns, there should always be a valid
+      // return value present. Instead of setting the return value here, we
+      // should have the ReturnValueSlot object set it beforehand.
+      if (!RetVal) {
+        RetVal = callOp.getResult();
+        // TODO(cir): Check for volatile return values.
+      }
+
+      // An empty record can overlap other data (if declared with
+      // no_unique_address); omit the store for such types - as there is no
+      // actual data to store.
+      if (dyn_cast<StructType>(RetTy) &&
+          cast<StructType>(RetTy).getNumElements() != 0) {
+        // NOTE(cir): I'm assuming we don't need to change any offsets here.
```

Comment:
Same as above, instead of handwaving, please try to figure out this in OG by making the path unrecheable + running tests, and then do the proper thing here.